### DISCUSSION TITLE
Using `SET LOCAL` with begin nested.

### DISCUSSION THREAD

**Discussion by michaelbukachi** [sqlalchemy/sqlalchemy#8249] [Usage Questions] [answered]
I've noticed using `SET LOCAL` affects the query beyond the nested transaction. Does `SET LOCAL` affect nested transactions or the current session? :thinking: 
```python
with db.begin_nested():
    try:
        db.execute(text("SET LOCAL statement_timeout TO 150;"))
        # Limit next query 150ms
        return db.scalar(select(....)))
    except OperationalError:
        db.rollback()
    # This query is affected
db.execute(...)
```

**Comment by zzzeek** [comment, marked as answer]
I have no idea what that is or what database or driver you're using.   anyway this would be something to do with the driver in use so check with them.

**Reply by michaelbukachi** [reply]
Sorry for this missing information. This is Postgres with psycopg2 driver. 
- [statment_timeout](https://postgresqlco.nf/doc/en/param/statement_timeout/)
- [Postgres SET](https://www.postgresql.org/docs/10/sql-set.html)

**Reply by zzzeek** [reply]
OK, so [here](https://www.postgresql.org/docs/current/sql-set.html)

> The effects of SET LOCAL last only till the end of the current transaction, whether committed or not

a "nested" transaction is just a SAVEPOINT, so that's not a "transaction" in PostgreSQL terms

**Reply by michaelbukachi** [reply]
Oh. I see :thinking: . Thanks for the clarification.
